qiime2/sdk/plugin_manager.py

```diff
@@ -350,15 +359,14 @@ def get_directory_format(self, semantic_type):
                 "Must provide a semantic type via `semantic_type`, not %r" %
                 semantic_type)
 
-        dir_fmt = None
-        for artifact_class_record in self.artifact_classes:
+        # Evan, ideally we could just lookup semantic_type in
+        # self.artifact_classes but properties get in the way. Is there a way
+        # to strip properties so this could be simplified to return
+        # self.artifact_classes[semantic_type] while catching a KeyError?
```

Comment:
There's not a global helper for this, but there is a `.duplicate()` method which allows you to override attributes of a type, so `(SomeType % Foo).duplicate(predicate=IntersectionExp())` would accomplish your goal. It's not pretty and it's also not recursive, so any variant fields would need to be recursively duplicated as well.

The behavior below is correct and the above sounds like too much work. So let's ignore this for now.